Components/CategoryResolver/AutoCategoryResolver.php

```diff
@@ -126,13 +122,17 @@ public function convertTreeToEntities(array $node, Category $parent = null, $cat
 
     /**
      * @param array $category
-     * @param Category $parent
+     * @param string $parentId
      * @return Category
      */
-    public function convertNodeToEntity(array $category, Category $parent)
+    public function convertNodeToEntity(array $category, $parentId)
     {
         $categoryModel = new Category();
         $categoryModel->fromArray($this->getCategoryData($category['name']));
+
+        $parent = $this->categoryRepository->findOneBy([
+            'id' => $parentId
```

Comment:
you can cast it to int. Useful when someone uses this function with string values. Doctrine won't cast automatically.
I had problems with varchar column in db and int values. then index doesn't work.

Reply:
If parent doesn't exist then it will throw an exception "Call getId on null", but I think it's fine